astropy/modeling/tests/test_fitters.py

```diff
@@ -234,10 +234,9 @@ def test_compound_model_raises_error(self):
     def test_chebyshev1D(self):
         """Tests fitting a 1D Chebyshev polynomial to some real world data."""
 
-        test_file = get_pkg_data_filename(os.path.join("data", "idcompspec.fits"))
-        with open(test_file) as f:
-            lines = f.read()
-            reclist = lines.split("begin")
+        test_file = Path(get_pkg_data_filename("data/idcompspec.fits"))
+        lines = test_file.read_text()
+        reclist = lines.split("begin")
```

Comment:
Does this fit in a single line?
```suggestion
        reclist = Path(get_pkg_data_filename("data/idcompspec.fits")).read_text().split("begin")
```

Reply:
If not yet, then when `get_pkg_data_filename` is upgraded to return a Path...

Reply:
I can start working on this in this PR if that makes it easier to review. My plan was to get the `get_pkg_data_*` methods work with pathlib natively in a new PR.

Reply:
No need to do it in this PR.

Reply:
I was just commenting that it will reduce the line length, maybe making it 1 line in the future.